### PR TITLE
Let cloud_admin do live migration

### PR DESCRIPTION
**Pull request by yiheqilin**
edit keystone policy.json to show Admin/instances in horizon to cloud_admin
edit nova policy.json to let cloud_admin trigger live migration

@Xiaohua-Shen Could you review this PR ?

### CODE REVIEW

**Comment by Xiaohua-Shen** [comment]
ok to test

**Comment by Xiaohua-Shen** [comment]
LGTM

File: roles/common/templates/etc/hosts [hunk 1]
```diff
@@ -29,7 +29,7 @@ ff02::2 ip6-allrouters
 
 {% if 'compute' in groups %}
 {% for host in groups['compute'] if inventory_hostname in groups['compute'] %}
-{{ hostvars[host].ansible_default_ipv4.address }} {{ hostvars[host].ansible_nodename }}
+{{ hostvars[host].ansible_default_ipv4.address }} {{ hostvars[host].ansible_nodename }} {{ hostvars[host].hostname_fqdn }}
```
Comment (nirajdp76) [inline roles/common/templates/etc/hosts]: We should not need this change. There is probably some other underlying root-cause for live migration to not work. I will be debugging this further tonight and provide more inputs.

**Comment by Xiaohua-Shen** [comment]
@lihkin213 the reason for admin_required_limit is:
In horizon, live migration action is under menu of  admin->instance->live migration action.
and dashboard to show admin->instance need privilege of 'admin_required' (that's hardcoded in dashboard).  Now cloud_admin has no privilege to show this panel.

So if we keep 'admin_required' unchanged and add 'cloud_admin' to this rule, then cloud_admin can see the panel , but also has all keystone related CRUD privilege.

What we tried to do is: 
let cloud_admin can see this panel: admin->instance->live migration action, and still has no other keystone privilege.

**Comment by lihkin213** [comment]
We should see if we can use horizon customization for this instead of modifying the policy file

File: roles/keystone/templates/etc/keystone/policy.json [hunk 2]
```diff
@@ -2,9 +2,11 @@
 
 {
 {% if policy_for_horizon|default('False')|bool %}
-    "admin_required": "role:admin or is_admin:1 or role:cloud_admin or role:project_admin or rule:owner",
+    "admin_required_limit": "role:admin or is_admin:1 or role:cloud_admin or role:project_admin or rule:owner",
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/policy.json]: I still do not see why this change to the policy file is required. cloud_admin has all the the Keystone CRUD privileges.

Comment (Xiaohua-Shen) [inline roles/keystone/templates/etc/keystone/policy.json]: for policy_for_horizon is false, "admin_required_limit" = "admin_required", and cloud_admin has CRUD, same as before.
for policy_for_horizon is true, "admin_required_limit" can do CRUD, but "admin_required" not.
cloud_admin has no CRUD

Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/policy.json]: So I verified that the change to the keystone policy.json is not required. You need to make sure that horizon customization is enabled as well as jellyroll. I could not test the actual live migration as I only had one compute host in my env. I was able to start the live migration and the api was invoked correctly.

**Comment by yiheqilin** [comment]
Base on previous reviews, edit nova policy only

**Comment by Xiaohua-Shen** [comment]
@lihkin213 verified on integration test env and it works.

**Comment by nirajdp76** [comment]
ok to test

**Comment by nirajdp76** [comment]
@lihkin213 can u provide your approval

File: roles/nova-common/templates/etc/nova/policy.json [hunk 3]
```diff
@@ -289,7 +289,7 @@
     "os_compute_api:os-admin-password": "rule:admin_or_owner",
     "os_compute_api:os-admin-password:discoverable": "@",
     "os_compute_api:os-aggregates:discoverable": "@",
-    "os_compute_api:os-aggregates:index": "rule:admin_api",
```
Comment (Xiaohua-Shen) [inline roles/nova-common/templates/etc/nova/policy.json]: add this is for command: "openstack hypervisor show compute-0".
Since for data migration, cloud_admin may want to view the hypervisor status to determine the target hypervisor.
https://docs.openstack.org/developer/nova/sample_policy.html said this policy is related to " List all aggregates". So is it ok to let cloud_admin have this privilege?

File: roles/nova-common/templates/etc/nova/policy.json [hunk 4]
```diff
@@ -106,7 +106,7 @@
     "compute_extension:admin_actions:resetNetwork": "rule:admin_or_owner",
     "compute_extension:admin_actions:injectNetworkInfo": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:admin_actions:createBackup": "rule:admin_or_owner",
-    "compute_extension:admin_actions:migrateLive": "rule:admin_api",
```
Comment (Xiaohua-Shen) [inline roles/nova-common/templates/etc/nova/policy.json]: This is to control if related action is shown in horizon. I think we need this for cloud_admin

File: roles/nova-common/templates/etc/nova/policy.json (hunk 3 above)
Comment (lihkin213) [inline roles/nova-common/templates/etc/nova/policy.json]: ok. I think that is okay.

**Comment by lihkin213** [comment]
retest

**Comment by lihkin213** [comment]
ok to test

**Comment by Xiaohua-Shen** [comment]
retest
